src/main/scala/chisel3/util/Lookup.scala

```diff
@@ -4,6 +4,29 @@ package chisel3.util
 
 import chisel3._
 
+/** For each element in a list, muxes (looks up) between cases (one per list element) based on a
+  * common address.
+  *
+  * @note This appears to be an odd, specialized operator that we haven't seen used much, and seems
+  *       to be a holdover from chisel2. This may be deprecated and removed, usage is not
+  *       recommended.
+  *
+  * @param addr common select for cases, shared (same) across all list elements
+  * @param default default value for each list element, should the address not match any case
+  * @param mapping list of cases, where each entry consists of a BitPat (compared against addr) and
+  *                a list of elements (same length as default) that is the output value for that
+  *                element (will have the same index in the output).
+  *
+  * @example {{{
+  *   ListLookup(1.U,  // address for comparison
+  *       List(      00.U, 01.U, 02.U)   // default "row" if none of the following cases match
+  *       List(1.U, (10.U, 11.U, 12.U),  // this "row" hardware-selected based off address 1.U
+  *            2.U, (20.U, 21.U, 22.U)
+  *       )
+  *   )  // hardware-evaluates to List(10.U, 11.U, 12.U)
+  *   // Note: if given address 0.U, the above would hardware evaluate to List(00.U, 01.U, 02.U)
```

Comment:
Not sure the types in this example are fully correct. I tried the following and it appears to work.

(Fun fact, `00.U` appears to be illegal)

```scala
ListLookup(2.U,  // address for comparison
                              List(10.U, 11.U, 12.U),   // default "row" if none of the following cases match
         Array(BitPat(2.U) -> List(20.U, 21.U, 22.U),  // this "row" hardware-selected based off address 2.U
               BitPat(3.U) -> List(30.U, 31.U, 32.U)
         )
     )
// hardware-evaluates to List(20.U, 21.U, 22.U)
```

Reply:
Fixed to your example. I guess that's what I get for not testing my code...